util/src/builder/embed/mod.rs

```diff
@@ -0,0 +1,429 @@
+    ///
+    /// # Examples
+    ///
+    /// Set the thumbnail to an image attachment with the filename
+    /// `"twilight.png"`:
+    ///
+    /// ```
+    /// # fn main() -> Result<(), Box<dyn std::error::Error>> {
+    /// use twilight_util::builder::embed::{EmbedBuilder, ImageSource};
+    ///
+    /// let embed = EmbedBuilder::new()
+    ///     .description("a picture of twilight")
+    ///     .thumbnail(ImageSource::attachment("twilight.png")?)
+    ///     .validate()?
+    ///     .build();
+    /// # Ok(()) }
+    /// ```
+    #[allow(clippy::missing_const_for_fn)]
+    pub fn thumbnail(mut self, image_source: ImageSource) -> Self {
+        self.0.thumbnail = Some(EmbedThumbnail {
+            height: None,
+            proxy_url: None,
+            url: image_source.0,
+            width: None,
+        });
+
+        self
+    }
+
+    /// Set the ISO 8601 timestamp.
+    pub const fn timestamp(mut self, timestamp: Timestamp) -> Self {
+        self.0.timestamp = Some(timestamp);
+
+        self
+    }
+
+    /// Set the title.
+    ///
+    /// Refer to [`TITLE_LENGTH`] for the maximum number of UTF-16 code points
+    /// that can be in a title.
+    ///
+    /// # Examples
+    ///
+    /// Set the title to "twilight":
+    ///
+    /// ```
+    /// # fn main() -> Result<(), Box<dyn std::error::Error>> {
+    /// use twilight_util::builder::embed::EmbedBuilder;
+    ///
+    /// let embed = EmbedBuilder::new()
+    ///     .title("twilight")
+    ///     .url("https://github.com/twilight-rs/twilight")
+    ///     .validate()?
+    ///     .build();
+    /// # Ok(()) }
+    /// ```
+    ///
+    /// [`TITLE_LENGTH`]: twilight_validate::embed::TITLE_LENGTH
+    pub fn title(mut self, title: impl Into<String>) -> Self {
+        self.0.title = Some(title.into());
+
+        self
+    }
+
+    /// Set the URL.
+    ///
+    /// # Examples
+    ///
+    /// Set the URL to [twilight's repository]:
+    ///
+    /// ```
+    /// # fn main() -> Result<(), Box<dyn std::error::Error>> {
+    /// use twilight_util::builder::embed::{EmbedBuilder, EmbedFooterBuilder};
+    ///
+    /// let embed = EmbedBuilder::new()
+    ///     .description("twilight's repository")
+    ///     .url("https://github.com/twilight-rs/twilight")
+    ///     .validate()?
+    ///     .build();
+    /// # Ok(()) }
+    /// ```
+    ///
+    /// [twilight's repository]: https://github.com/twilight-rs/twilight
+    pub fn url(mut self, url: impl Into<String>) -> Self {
+        self.0.url = Some(url.into());
+
+        self
+    }
+}
+
+impl Default for EmbedBuilder {
+    /// Create an embed builder with a default embed.
+    ///
+    /// All embeds have a "rich" type.
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl TryFrom<EmbedBuilder> for Embed {
+    type Error = EmbedValidationError;
+
+    /// Convert an embed builder into an embed, validating its contents.
+    ///
+    /// This is equivalent to calling [`EmbedBuilder::build`].
```

Comment:
No it's not equivalent to calling build, it also validates.

Reply:
a lot of these are unchanged from the old builder, thanks for catching them.